### PR TITLE
refactor(v2): improve semantic doc sidebar markup

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

Just make the markup for the doc sidebar a little more semantic.

Refs:

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside
- https://fae.disability.illinois.edu/rulesets/ROLE_7/
- https://html.com/tags/aside/
- http://mariechatfield.com/blog/semantic-html 

For example, Bootstrap, VuePress, Next.js sites use similar markup.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Preview (no visual changes, actually)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx [hunk 1]
```diff
@@ -344,11 +344,17 @@ function DocSidebar({
             [styles.menuWithAnnouncementBar]:
               !isAnnouncementBarClosed && showAnnouncementBar,
           },
-        )}>
+        )}
+        aria-label={translate({
+          id: 'theme.docs.sidebar.menuLabel',
+          message: 'Docs navigation',
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx:350]: I just thought that maybe we shouldn't use the word "docs" in ARIA labels? After all, it won't always be exactly docs, right? Maybe should use "Secondary navigation" or something like that?

Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx:350]: Gatsby has "Secondary navigation" so why not. Isn't there value to also mention it's a sidebar? like `Secondary navigation (sidebar)` or something?

**Review by slorber — CHANGES_REQUESTED** [review]
I'd prefer if all aria label keys contain `AriaLabel` by convention and for consistency

File: packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx [hunk 2]
```diff
@@ -344,11 +344,17 @@ function DocSidebar({
             [styles.menuWithAnnouncementBar]:
               !isAnnouncementBarClosed && showAnnouncementBar,
           },
-        )}>
+        )}
+        aria-label={translate({
+          id: 'theme.docs.sidebar.menuLabel',
```
[the remark this answers is not in the document]
Comment (slorber) [reply packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx:349]: what about `navAriaLabel ` for consistency with the blog?

[the remark this answers is not in the document]
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx]: Done

File: packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx (hunk 1 above)
Comment (lex111) [reply packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx]: Propose simply "Sidebar navigation".

**Comment by slorber** [comment]
👍

**Pull request merged**